caas/kubernetes/provider/k8s.go

```diff
@@ -125,6 +136,86 @@ func newK8sConfig(cloudSpec environs.CloudSpec) (*rest.Config, error) {
 	}, nil
 }
 
+// Config returns environ config.
+func (k *kubernetesClient) Config() *config.Config {
+	return k.envCfg.Config
```

Comment:
You need to use the mutex and return a copy of the config.
See ec2/environ.go or vsphere/environ.go etc